docs/how-tos/validate-nexus-file.md

```diff
@@ -0,0 +1,802 @@
+                 For Raman spectroscopy or ellipsometry use the respective specializations
+                 of NXoptical_spectroscopy.
+```
+
+or like this, if the respective entry is not found in the defintion:
+
+```
+DEBUG: ===== ATTRS (//entry/instrument/software_RC2/program@url)
+DEBUG: value: https://www.jawoollam.com/ellipsometry-software/completeease 
+DEBUG: classpath: ['NXentry', 'NXinstrument']
+DEBUG: NOT IN SCHEMA
+DEBUG:
+```
+
+The first example was for for "experiment\_type" entry in the "NXoptical\_spectroscopy" definition.
+
+The second example was for the "software\_TYPE" attribute @URL entry in the "NXoptical\_spectroscopy" definition. Here the problem was that "url" was used instead of "URL".
+
+
+
+# 1.2.1 Installation of read_nexus
+
+This is installed with pip:
+
+```
+pip install pynxtools
+```
+
+
+# 1.2.2 Using the read_nexus function
+
+Open your terminal. Assuming there is a folder at:
+
+For Linux:
+
+```
+/home/USER/nexusvalidation
+```
+
+For Windows:
+
+```
+C:\nexusvalidation
+```
+
+Put into this folder your NeXus file, for example the [Raman.nxs file](https://zenodo.org/records/13373909/files/Raman.nxs?download=1).
+
+Then use:
+
+```
+read_nexus -f C:\nexusvalidation\Raman.nxs
+```
+
+shows the output like this:
+
+```
+===== FIELD (//entry/data/spectrum_data_y): <HDF5 dataset "spectrum_data_y": shape (1600,), type "<f8">
+DEBUG: ===== FIELD (//entry/data/spectrum_data_y): <HDF5 dataset "spectrum_data_y": shape (1600,), type "<f8">
+value: [ 288.5499878  289.         288.4500122 ... 1875.        1889.349976 ...
+DEBUG: value: [ 288.5499878  289.         288.4500122 ... 1875.        1889.349976 ...
+Dataset referenced as NXdata SIGNAL
+DEBUG: Dataset referenced as NXdata SIGNAL
+===== ATTRS (//entry/data/spectrum_data_y@long_name)
+DEBUG: ===== ATTRS (//entry/data/spectrum_data_y@long_name)
+value: Raman Intensity 
+DEBUG: value: Raman Intensity 
+Dataset referenced as NXdata SIGNAL
+DEBUG: Dataset referenced as NXdata SIGNAL
+===== ATTRS (//entry/data/spectrum_data_y@units)
+DEBUG: ===== ATTRS (//entry/data/spectrum_data_y@units)
+value: counts 
+DEBUG: value: counts 
+
+DEBUG: 
+For Axis #0, 1 axes have been identified: [<HDF5 dataset "spectrum_data_x_Raman": shape (1600,), type "<f8">]
+DEBUG: For Axis #0, 1 axes have been identified: [<HDF5 dataset "spectrum_data_x_Raman": shape (1600,), type "<f8">]
+```
+
+Search for fields which are not found in the NeXus definition by searching for the line: "DEBUG: NOT IN SCHEMA". Recheck the used NeXus definition to eliminate the problem. Be careful with upper and lower case notation and correct spelling.
+
+Keep in mind, that the output provides quite some information. This is useful for software development, but may be a bit too much for validation purposes.
+
+Similar features as the tables and messages provided from punx and cnxvalidate for pynxtools are [planned to be implemented in the future](<https://github.com/FAIRmat-NFDI/pynxtools/pull/333>).
```

Comment:
Maybe remove? We have this covered above.

Reply:
yea, will remove it. Was from my first draft

Reply:
removed